File: CLIPS_Series/2023-03-02-Integrating CLIPS into C and C# with CLIPS Connectors.cs
using CLIPSNET;

namespace MyProgram {
    class Program {
        static void Main(string[] args) {
            /* Create a new CLIPS environment */
            Environment theEnv = new Environment();
            /* Load the rules */
            theEnv.Load("my-rules.clp");

            /* Define a fact */
            theEnv.AssertString("(my-fact)");

            /* Run the rules */
            theEnv.Run();

            /* Destroy the environment */
            theEnv.Destroy();
        }
    }
}

File: CLIPS_Series/2023-03-02-Integrating CLIPS into C and C# with CLIPS Connectors.c
#include "clips.h"

int main() {
void *theEnv;
theEnv = CreateEnvironment();

/* Load the rules */
Load(theEnv, "my-rules.clp");

/* Define a fact */
AssertString(theEnv, "(my-fact)");

/* Run the rules */
Run(theEnv, -1);

/* Destroy the environment */
DestroyEnvironment(theEnv);

return 0;
}